modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/messages/StatisticsGatheringRequest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.query.stat.messages;
+
+import org.apache.ignite.internal.GridDirectMap;
+import org.apache.ignite.internal.managers.communication.GridIoPolicy;
+import org.apache.ignite.plugin.extensions.communication.Message;
+import org.apache.ignite.plugin.extensions.communication.MessageCollectionItemType;
+import org.apache.ignite.plugin.extensions.communication.MessageReader;
+import org.apache.ignite.plugin.extensions.communication.MessageWriter;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Request to gather statistics message.
+ */
+public class StatisticsGatheringRequest implements Message {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** */
+    public static final short TYPE_CODE = 179;
+
+    /** Gathering id. */
+    private UUID gatId;
+
+    /** Request id. */
+    private UUID reqId;
+
+    /** Keys to partitions to gather statistics by. */
+    @GridDirectMap(keyType = StatisticsKeyMessage.class, valueType = int[].class)
+    private Map<StatisticsKeyMessage, int[]> keys;
```

Comment:
StatisticsKeyMessage has default equals and hashcode methods. IS it OK?

Reply:
No, it's not, since it used here: https://github.com/gridgain/gridgain/pull/1660/files#diff-f727e80fc5250f851bb87b1f9cd9b5c159e5ca0cf094623418d515211784c82eR459

Reply:
HashCode&Equals implemented